internal/addon/addon.go

```
@@ -116,83 +188,240 @@ func AgentHealthProber() *agent.HealthProber {
+					return fmt.Errorf("%w: %s with key %s", errProbeValueIsNil, identifier.Resource, identifier.Name)
+				}
+
+				if *value.Value.String != "True" {
+					return fmt.Errorf("%w: %s status condition type is %s for %s", errProbeConditionNotSatisfied, identifier.Resource, *value.Value.String, identifier.Name)
+				}
+				// uiplugin passes the health check
+			}
+		case "customresourcedefinitions":
+			switch addoncfg.Name {
+			case "scrapeconfigs.monitoring.rhobs":
+				if err := checkScrapeConfigCRD(logger, field.FeedbackResult.Values, mc); err != nil {
+					return fmt.Errorf("%w: %s with key %s", err, identifier.Resource, identifier.Name)
+				}
+			}
+			continue
+		default:
+			// If a resource is being probed it should have a health check defined
+			return fmt.Errorf("%w: %s with key %s/%s", errUnknownResource, identifier.Resource, identifier.Namespace, identifier.Name)
+		}
+	}
+	return nil
+}
+
+func checkScrapeConfigCRD(logger logr.Logger, feedbackValues []workv1.FeedbackValue, mc *v1.ManagedCluster) error {
+	if len(feedbackValues) == 0 {
+		return errProbeValueIsNil
+	}
+
+	var version, isEstablished string
+	for _, value := range feedbackValues {
+		switch value.Name {
+		case addoncfg.PrometheusOperatorVersionFeedbackName:
+			if value.Value.String != nil {
+				version = *value.Value.String
+			}
+		case addoncfg.IsEstablishedFeedbackName:
+			if value.Value.String != nil {
+				isEstablished = *value.Value.String
+			}
+		}
+	}
+
+	if strings.ToLower(isEstablished) != "true" {
+		return fmt.Errorf("%w: resource is not established", errProbeConditionNotSatisfied)
+	}
+
+	if version == "" {
+		return fmt.Errorf("%w: prometheus operator version not found in scrapeconfigs.monitoring.rhobs CRD", errProbeConditionNotSatisfied)
+	}
+
+	isOlder, err := isVersionOlder(version, minPrometheusOperatorVersion)
+	if err != nil {
+		logger.Info("failed to parse prometheus operator version", "error", err.Error(), "managedCluster", mc.Name)
```

shouldn't we return the err here as well? :thinking: